### PR TITLE
[FLINK-28862][python][format/parquet] Support ParquetRowDataWriter

### PR DESCRIPTION
**Pull request by Vancior**
## What is the purpose of the change

This PR supports `ParquetRowDataWriter.for_row_type` API in PyFlink, to create a `BulkWriterFactory` that writes `Row` records into Parquet files in a batch fashion.

## Verifying this change

This change added tests and can be verified as follows:

- `FileSinkParquetRowDataWriterTests` in test_parquet.py

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (yes)
  - If yes, how is the feature documented? (Python Sphinx doc)

### CODE REVIEW

File: flink-python/pyflink/fn_execution/coders.py [hunk 1]
```diff
@@ -725,13 +725,17 @@ def from_type_info_proto(type_info):
             return RowCoder(
                 [from_type_info_proto(f.field_type) for f in type_info.row_type_info.fields],
                 [f.field_name for f in type_info.row_type_info.fields])
-        elif field_type_name == type_info_name.PRIMITIVE_ARRAY:
+        elif field_type_name in (
+            type_info_name.PRIMITIVE_ARRAY,
```
Comment (dianfu) [inline flink-python/pyflink/fn_execution/coders.py:729]: What's the purpose of this change?

File: flink-python/pyflink/datastream/formats/parquet.py [hunk 2]
```diff
@@ -155,3 +158,54 @@ def for_generic_record(schema: 'AvroSchema') -> 'BulkWriterFactory':
...
+            ...     ['string', 'int_array'],
+            ...     [Types.STRING(), Types.LIST(Types.INT())]
+            ... )
+            >>> sink = FileSink.for_bulk_format(
+            ...     OUTPUT_DIR, ParquetRowDataWriter.for_row_type(
+            ...         row_type,
+            ...         hadoop_config=Configuration(),
+            ...         utc_timestamp=True,
+            ...     )
+            ... ).build()
```
[the remark this answers is not in the document]
Comment (Vancior) [reply flink-python/pyflink/datastream/formats/parquet.py:195]: Why no `build` ?

File: flink-python/pyflink/fn_execution/coders.py (hunk 1 above)
Comment (Vancior) [reply flink-python/pyflink/fn_execution/coders.py:729]: This fixes an existing bug of list deserialization.

**Review by dianfu — APPROVED** [review]
